Jormungandr: Ability to call HERE routing service to get realtime car routing data

antoine-de opened the pull request:
Add another streetnetwork connector calling [HERE](https://developer.here.com) service

the nice feature of this service is that there is realtime data on the streetnetwork.

One of the down side is that the routing matrix is limited to 100 elements, so it is usable only in a very sparse dataset (like the SNCF).

This is a first rough implementation.

some stuff needs to be implemented (but I'm not sure they are mandatory):
* search range (but the here api limit the range as a distance and we do it as a travel time)
* better directions (for the moment there is not the name of the streets and the direction to take)

what do cannot do:
* change the vehicle speed (only the walking speed can be set)
* the end fallback will have a false datetime as we cannot be precise (it depends on the public transport taken)

antoine-de committed:
jormungandr: first here implementation

antoine-de committed:
integration tests

antoine-de committed:
handle the datetime of the query

kinnou02 commented on source/jormungandr/jormungandr/street_network/here.py:
```diff
@@ -0,0 +1,291 @@
+        try:
+            return self.breaker.call(requests.get, url, timeout=self.timeout, params=params)
+        except pybreaker.CircuitBreakerError as e:
+            log.error('Valhalla routing service dead (error: {})'.format(e))
```
please add an extra parameters with the id of the system

antoine-de committed:
better handling of newrelic

antoine-de committed:
departure/arrival time handling

antoine-de committed:
revert clever arrival handling

as arrival constraint is available only for PT transport in HERE, so we
can't use it

antoine-de committed:
display base time

antoine-de committed:
better json logs

antoine-de committed:
add timeout test
